### PR TITLE
Fix bug in releasing buffer for Async GetWorkflowExecutionHistory

### PR DESCRIPTION
**Pull request by meiliang86**

### CODE REVIEW

File: src/main/java/com/uber/cadence/client/WorkflowOptions.java [hunk 1]
```diff
@@ -124,9 +124,10 @@ public Builder setWorkflowId(String workflowId) {
      * conditions Cadence allows two workflows with the same domain and workflow id run
      * simultaneously.
      * <li>
-     * <ul>
-     *   AllowDuplicateFailedOnly is a default value. It means that workflow can start if
-     *   previous run failed or was cancelled or terminated.
+     *
+     *     <ul>
```
Comment (vancexu) [inline src/main/java/com/uber/cadence/client/WorkflowOptions.java:128]: nit: indent?

Comment (meiliang86) [reply src/main/java/com/uber/cadence/client/WorkflowOptions.java:128]: This is somehow changed automatically by the code formatter through gradlew build.

**Comment by meiliang86** [comment]
> Will other API have similar issues?

No. We only implemented one Async API which is this one.

File: src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java [hunk 2]
```diff
@@ -2336,18 +2336,17 @@ public void GetWorkflowExecutionHistory(
...
-                return;
-              })
-          .exceptionally(
-              (e) -> {
-                log.error("Unexpected error in GetWorkflowExecutionHistory", e);
-                return null;
-              });
-    } finally {
-      if (response != null && response.isDone()) {
-        response.join().release();
```
Comment (mkolodezny) [inline src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java:2348]: nit: so the response is never released if not done immediately here?

Comment (meiliang86) [reply src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java:2348]: That's the bug to be fixed in this diff, right?

**Comment by meiliang86** [comment]
Buildkite passed: https://buildkite.com/uberopensource/cadence-java-client/builds/374

**Pull request merged**